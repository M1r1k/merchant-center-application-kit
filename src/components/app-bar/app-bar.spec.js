import React from 'react';
import { shallow } from 'enzyme';
import FetchUser from '../fetch-user';
import ProjectSwitcher from '../project-switcher';
import AppBar from './app-bar';

describe('rendering', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AppBar match={{ params: { projectKey: 'test-1' } }} />);
  });
  it('should match layout structure', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render element for locale switcher portal', () => {
    expect(wrapper).toRender('#locale-switcher');
  });
  describe('<ProjectSwitcher>', () => {
    let fetchUserChildrenWrapper;
    describe('when user is loading', () => {
      beforeEach(() => {
        fetchUserChildrenWrapper = shallow(
          <div>
            {wrapper.find(FetchUser).prop('children')({ isLoading: true })}
          </div>
        );
      });
      it('should not render <ProjectSwitcher>', () => {
        expect(fetchUserChildrenWrapper).not.toRender(ProjectSwitcher);
      });
    });
    describe('when user is not loading', () => {
      describe('when user has available projects', () => {
        beforeEach(() => {
          fetchUserChildrenWrapper = shallow(
            <div>
              {wrapper.find(FetchUser).prop('children')({
                isLoading: false,
                user: { availableProjects: [{ key: 'p1' }, { key: 'p2' }] },
              })}
            </div>
          );
        });
        it('should render <ProjectSwitcher>', () => {
          expect(fetchUserChildrenWrapper).toRender(ProjectSwitcher);
        });
        it('should pass projectKey to <ProjectSwitcher>', () => {
          expect(fetchUserChildrenWrapper.find(ProjectSwitcher)).toHaveProp(
            'projectKey',
            'test-1'
          );
        });
        it('should pass availableProjects to <ProjectSwitcher>', () => {
          expect(fetchUserChildrenWrapper.find(ProjectSwitcher)).toHaveProp(
            'availableProjects',
            [{ key: 'p1' }, { key: 'p2' }]
          );
        });
      });
      describe('when user does not have available projects', () => {
        beforeEach(() => {
          fetchUserChildrenWrapper = shallow(
            <div>
              {wrapper.find(FetchUser).prop('children')({
                isLoading: false,
                user: { availableProjects: [] },
              })}
            </div>
          );
        });
        it('should not render <ProjectSwitcher>', () => {
          expect(fetchUserChildrenWrapper).not.toRender(ProjectSwitcher);
        });
      });
    });
  });
});
