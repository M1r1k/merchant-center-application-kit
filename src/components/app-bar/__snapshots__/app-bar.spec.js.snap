// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match layout structure 1`] = `
<header
  className="top-navigation"
  data-test="top-navigation"
>
  <div
    className="logo"
  >
    <img
      alt="Logo"
      className="logo-img"
      src="test-file-stub"
    />
  </div>
  <div
    className="loader-container"
    id="loading-spinner"
  />
  <div
    className="navigation"
  >
    <Inline
      alignItems="center"
      scale="m"
    >
      <div
        className="switchers"
      >
        <Inline
          alignItems="center"
          scale="s"
        >
          <div
            id="locale-switcher"
          />
          <FetchUser />
        </Inline>
      </div>
      <div
        className="spacer"
      />
      <Inline
        alignItems="center"
        scale="s"
      >
        <UserSettingsMenu />
      </Inline>
    </Inline>
  </div>
</header>
`;
